app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -62,20 +63,21 @@ class CiYaml {
   /// Filter [targets] to only those that are expected to run for [branch].
   ///
   /// A [Target] is expected to run if:
-  ///   1. [Target.enabledBranches] exists and contains [branch].
-  ///   2. Otherwise, [config.enabledBranches] contains [branch].
+  ///   1. [Target.enabledBranches] exists and matches [branch].
+  ///   2. Otherwise, [config.enabledBranches] matches [branch].
   List<Target> _filterEnabledTargets(Iterable<Target> targets) {
     final List<Target> filteredTargets = <Target>[];
 
     // 1. Add targets with local definition
     final Iterable<Target> overrideBranchTargets =
         targets.where((Target target) => target.value.enabledBranches.isNotEmpty);
     final Iterable<Target> enabledTargets =
-        overrideBranchTargets.where((Target target) => target.value.enabledBranches.contains(branch));
+        overrideBranchTargets.where((Target target) => RegExp(target.value.enabledBranches.join('|')).hasMatch(branch));
```

Comment:
I suggest pulling this into a helper method, both to help debugging when inevitably we fail here, and so we can unit test it.

Reply:
also note turning `'master'` into a regex means it will match `'my-feature-branch-off-of-master'`